### DISCUSSION TITLE
What about Vue?

### DISCUSSION THREAD

**Discussion by DavidVollmers** [tailwindlabs/headlessui#3183] [General]
Now with the release of [@headlessui/react@v2.0.0](https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v2.0.0) I am wondering what is in plan for @headlessui/vue? Will there be feature congruency?

**Comment by medz** [comment]
Floating ui supports Vue, so I believe building consistency is important. I really look forward to quickly opening `@headlessui/vue` v2

**Reply by philngo** [reply]
The [React Aria](https://react-spectrum.adobe.com/react-aria/hooks.html) dependency may be harder to replicate exactly in Vue.

**Comment by thalesagapito** [comment]
I'd also point out that for any future releases it would be nice to get this sort of info right out of the gate. It feels weird to me that a library that supports both Vue and React has no mention of Vue anywhere on the official release post.

**Reply by LucaRed** [reply]
To be fair, the blog post is called "[Headless UI v2.0 **for React**](https://tailwindcss.com/blog/headless-ui-v2)", but I agree it's a bit weird to see no mention of Vue at all.

I hope Vue support is coming soon, because there's a big Laravel community that loves Vue and this library is top notch :ok_hand:

**Comment by philngo** [comment]
The blog post introducing the catalyst preview from Dec 2023 ("[Introducing Catalyst: A modern UI kit for React](https://tailwindcss.com/blog/introducing-catalyst)"), did briefly mention Vue.

> These improvements are React-only for now during this early alpha period, but we plan to bring all of these improvements to Vue as well before tagging v2.0.

**Reply by DavidVollmers** [reply]
Well that didn't happen 😄

**Reply by philngo** [reply]
I'm guessing they meant before tagging @headlessui/vue@v2.0.0

**Comment by cmcmanus96** [comment]
I feel bad "complaining" because I know the Tailwind team is releasing banger after banger, and React users are certainly eating good right now, but it would definitely just be nice to know if Vue is even still on the roadmap for the future. I honestly don't even care about a date! I think I just have some (probably irrational) fear that Vue is being abandoned for releases going forward so the team can focus building for React. I don't imagine having to support & maintain two flavors of the library is ideal. Those guys certainly don't owe us anything, but _any_ communication would be appreciated. I have a losing record on getting a response when I ask on Twitter 😅

**Reply by DavidVollmers** [reply]
Well they certainly owe any paying tailwindui customer who relies on VueJS an answer and potentially more...

**Reply by cmcmanus96** [reply]
Well, to play devils advocate, you could call Tailwind UI a finished product. While Tailwind UI _does_ use Headless UI for the components, they've already delivered on the product (and even added more, for free, since initial launch!). We're not losing anything, we're just also not gaining a 1:1 match with React on the Headless UI side.

**Reply by DavidVollmers** [reply]
With the catalyst UI kit which is part of tailwindui this looks different since it is using the new react features.

**Comment by boutzamat** [comment]
I was super impressed by Catalyst, and decided to use it for my next (current project) - until i started looking into the docs, and noticed that this template uses Headless UI v2. Didn't even notice there was a v2 out, but i thought "Nice, a newer version of Headless UI!" - until i noticed that it's currently for React only. Kind of bummed and disappointed as a Vue user that Vue is barely mentioned, and that it's not getting the same love as React.
I am currently trying to recreate the theme of Catalyst using the classic Headless UI 1 but i don't feel good about it.
I hope that Vue gets Headless UI v2 soon, or at least any kind of mentioning, so we know that it's on its way.

**Comment by ayhanfuat** [comment]
Looks like it is not going to happen soon. From Adam's [Twitter](https://x.com/adamwathan/status/1807502683189772362):

> Hey! No concrete plans to do Catalyst for Vue but not something I'm actually against, just hasn't earned a spot on the roadmap. Headless UI v2 for Vue I just don't have a good answer for — I think we should do it but our plate is already full for the rest of the year with Tailwind CSS v4.0 related projects so I just don't really know yet what we're going to do.

**Comment by SoCuul** [comment]
@RobinMalfait Does `@headlessui/vue` 2.0 support have an estimated release date, or at least a spot on the roadmap?

**Comment by aa-ndrej** [comment]
@ico85 a good alternative is [Reka UI](https://reka-ui.com). When I started using HeadlessUI, Radix Vue [^1] was new and not quite ready yet. But today, it definitely is! It has way more components and is actively maintained.
Hope this helps. 🙏🏼

[^1]: Prior to v2 the library was called [Radix Vue](https://www.radix-vue.com)

**Reply by davidmyersdev** [reply]
I used to recommend Headless UI as the best option for Vue, but Reka UI (apparently renamed) looks like it might become my new default!

**Reply by aa-ndrej** [reply]
Same! And that is right, they renamed the library to Reka UI for v2. I updated my comment accordingly, thanks!

**Reply by LucaRed** [reply]
The good part of Headless UI is that it's integrated with [Tailwind Plus](https://tailwindcss.com/plus) [^1], so it's easy to copy-paste all the components you need. That's why I'm so sad at seeing Vue support dying out.

[^1]: Previously called "Tailwind UI"

**Reply by SoCuul** [reply]
I've personally started [NuxtUI](https://ui.nuxt.com/) and have been really enjoying it. They also recently switched from HeadlessUI to Reka UI as a base, with the recent v3 release.

**Comment by weirdlooop** [comment]
I am switching to https://reka-ui.com/

**Reply by dsuryadi** [reply]
same here!

**Comment by mbachmann97** [comment]
Vue please.. 🥺

**Comment by heydayle** [comment]
Switch to another library, they almost forgot to update Vue. 
I recommend the following libraries: [NuxtUI (Nuxt Official)](https://ui.nuxt.com), [Reka-ui (base on Vue)](https://reka-ui.com),,... these libraries support vue very well.

Bonus: [Inspira-ui](https://inspira-ui.com/), [VueBits](https://vue-bits.dev)

**Reply by weirdlooop** [reply]
Reka ui is great, it is actually better than headlessui, its a shame as i will not buy any tailwind products because of this.
